2018-05-05  Christina Oliveira  <oliveira@lnxsrv07.seas.ucla.edu>

	* lib/timespec.h (timespec_cmp):
	Fixes a bug regarding time sorting, where old files were presented as new files.
	This fix removes subtraction,thereofre avoiding the problem overflow, and replaces it with a comparison.

--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec.h	2018-05-05 11:00:09.900872000 -0700
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec > b.tv_sec) {
+    return 1;
+  }
+ else if (a.tv_sec < b.tv_sec) {
+    return -1;
+  }
+  return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP






